models/player.go

```diff
@@ -312,6 +312,56 @@ func GetPlayerDetails(db DB, encryptionKey []byte, gameID, publicID string) (map
 	return result, nil
 }
 
+// GetPlayersToEncrypt get players that have plain text name
+func GetPlayersToEncrypt(db DB, encryptionKey []byte, amount int) ([]*Player, error) {
+	query := `SELECT p.*
+	FROM players p
+		LEFT JOIN encrypted_players ep ON p.id = ep.player_id
+	WHERE ep.player_id IS NULL
+	LIMIT $1`
+
+	var players []*Player
+	_, err := db.Select(&players, query, amount)
+	if err != nil {
+		return nil, err
+	}
+
+	for _, player := range players {
+		decryptedName, err := util.DecryptData(player.Name, encryptionKey)
```

Comment:
Is the player name encrypted by this point?

Reply:
Hmm nice catch... What happens if not? The script will try to decrypt a "never encrypted name", right?

Reply:
Hmm, how we are not using transactions (as you mention in another comment), the script is testing against the encryption to prevent that in somehow one encrypted player that was insert without EncryptedPlayer will be re-encrypted by the script that results in an error. In the normal case, the script tries to decrypt and throws an error that is ignored and the script goes ahead with your execution trying to decrypt other players.